### PR TITLE
Collections Endpoints 

### PR DESCRIPTION
**Pull request by matbroughty**
4 Collections Endpoints from the v1 [Twitter Collections API ](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/curate-a-collection/overview)

1. [Collections Create](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/curate-a-collection/api-reference/post-collections-create)
2. [Collections Curate](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/curate-a-collection/api-reference/post-collections-entries-curate)
3. [Collections Entries](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/curate-a-collection/api-reference/get-collections-entries)
4. [Collections Destroy](https://developer.twitter.com/en/docs/twitter-api/v1/tweets/curate-a-collection/api-reference/post-collections-destroy)

Not sure how widely used the collections api is but I use it a lot for the timstwitterlisteningparty.com site so thought I would share! 

Can add the other methods if of any use.

### CODE REVIEW

**Comment by redouane59** [comment]
Hi @matbroughty , thank you very much for this PR. I'll review it this week-end and i'll keep you informed !

**Comment by matbroughty** [comment]
Hello @redouane59 . Thanks for the library! Very easy to use and understand. Not sure how many people use the collections api (I do though!)  - it is a little odd compared to the other endpoints but if of any use and/or you would like changes let me know. Cheers. Mat

**Comment by redouane59** [comment]
Hi @matbroughty , i just watched your code and it has a really high quality, thank you so much for your work! I'm not really using these endpoints but that's a great opportunity for other users who may need it. 
I'm going to merge your PR now, everything will be available on the next release (1.17).

have a good week-end !

**Comment by redouane59** [comment]
Hi @matbroughty , are you sure that the `TimeLineOrder.CHRONOLOGICAL_REVERSE` and `TimeLineOrder.CHRONOLOGICAL` work ? Because creating collections with the two values, the two collections are exactly the same on tweetdeck
image

```
        twitterClient.collectionsCreate("Min likes & RT", "Min likes & RT", "min-likes-rt", TimeLineOrder.CHRONOLOGICAL);
        twitterClient.collectionsCreate("Min likes & RT R", "Min likes & RT R", "min-likes-rt-r", TimeLineOrder.CHRONOLOGICAL_REVERSE);
```

File: src/main/java/com/github/redouane59/twitter/dto/collections/TimeLineOrder.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * timeline_order used when creating a collection - https://api.twitter.com/1.1/collections/create.json
+ */
+public enum TimeLineOrder {
+
+    // oldest tweet first
+    CHRONOLOGICAL("tweet_chron"),
+    // order tweets are added to the collection (default)
+    CURATION_ORDER("curation_reverse_chron"),
+    // most recent tweet first
+    CHRONOLOGICAL_REVERSE("tweet_reverse_chron ");
```
Comment (redouane59) [inline src/main/java/com/github/redouane59/twitter/dto/collections/TimeLineOrder.java:13]: Hahaha fucking space...

Comment (matbroughty) [reply src/main/java/com/github/redouane59/twitter/dto/collections/TimeLineOrder.java:13]: Sorry @redouane59. Well found!

**Pull request merged**